sample-apps/README.md

```diff
@@ -38,3 +38,245 @@ List of sample apps across all repositories in [aws-observability](https://githu
+* **README** - The readme that will go into each sample app’s code base detailing how to setup the sample app (this should be same across all sample apps); although redundant, each repository should still have it to avoid looking across sample apps for a readme. This readme should also include the Non-conformance section specific to the sample app it belongs to.
+
+### Functional Requirements
+
+In a perfect world where everything is stable upstream, every sample app should emit the same Metrics, Logs, and Traces if given a specific routine (e.g. Python and Go sample apps both receive the same 3 requests → both emit the same signals in the same format). The following specifications will detail what exactly will be tracked as a metric log or trace. Each sample app should be configured with manual instrumentation from OTEL. Auto instrumentation can also be implemented.
+Metrics and traces should be OTLP and Logs should be in JSON format using any popular logging library for each respective language with the appropriate license. (e.g. Log4j for java, Logrus for Go)
+The model of how each sample app should function is displayed below:
+
+![Alt text](./sampleapp.png?raw=true "Sample App")
+
+**Metric type color code:** Counter (Green), Histogram (Orange), UpDownCounter (Blue), Gauge (Purple)
+The configuration file configures Host, Port, TimeAlive, TotalHeapSize, ThreadsActive, and CpuUsage. It will also configure TimeInterval. The following example YAML file demonstrates the structure and arbitrarily chosen default values for the previously mentioned configurable metrics.
+This configuration file should be set by an environment variable (SAMPLE_APP_CONF) or else it will default to the below snippet. If two sample apps are deployed by default, there will be an error for using the same port twice.
+
+```
+---
+Host: "0.0.0.0"                       # Host - String Address
+Port: "8080"                          # Port - String Port
+TimeInterval: 1                       # Interval - Time in seconds to generate new metrics
+RandomTimeAliveIncrementer: 1         # Metric - Amount to incremement metric by every TimeInterval
+RandomTotalHeapSizeUpperBound: 100    # Metric - UpperBound for TotalHeapSize for random metric value every TimeInterval
+RandomThreadsActiveUpperBound: 10     # Metric - UpperBound for ThreadsActive for random metric value every TimeInterval
+RandomCpuUsageUpperBound: 100         # Metric - UppperBound for CpuUsage for random metric value every TimeInterval
+SampleAppPorts: []                    # Sampleapp ports to make calls to   
+ResourceDetector: ''                  # String to specify resource detector
+```
+
+Every sample app will assume a configuration file with these variable names.
+The testing framework MUST not test against these Metrics due to upstream unstable SDK & API in certain languages. These are mainly for being able to demonstrate some type of metrics being generated in languages that support it. 
```

Comment:
What does this mean? What assertions cannot be made against these metrics? Can we make assertions that they exist with the correct metric type?